Makefile

```diff
@@ -150,15 +153,16 @@ generate-go: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Runs Go related generate tar
         paths=./cmd/clusterctl/api/...
 
 .PHONY: generate-bindata
-generate-bindata: $(GOBINDATA_GEN) ## generate code for embedding the clusterctl api manifest
-	# package manifest YAML into a single file
-	mkdir -p ./cmd/clusterctl/config/manifest/
-	kubectl kustomize ./cmd/clusterctl/config/crd > ./cmd/clusterctl/config/manifest/clusterctl-api.yaml
-	# generate go-bindata, add boilerplate, then cleanup
-	go generate ./cmd/clusterctl/config/generate.go
-	cat ./hack/boilerplate/boilerplate.generatego.txt ./cmd/clusterctl/config/crd_manifests.go > ./cmd/clusterctl/config/manifest/crd_manifests.go
-	cp ./cmd/clusterctl/config/manifest/crd_manifests.go ./cmd/clusterctl/config/crd_manifests.go
-	rm -r ./cmd/clusterctl/config/manifest/
+generate-bindata: $(GOBINDATA) clean-bindata ## Generate code for embedding the clusterctl api manifest
+	# Package manifest YAML into a single file.
+	mkdir -p $(GOBINDATA_CLUSTERCTL_DIR)/manifest/
+	kubectl kustomize $(GOBINDATA_CLUSTERCTL_DIR)/crd > $(GOBINDATA_CLUSTERCTL_DIR)/manifest/clusterctl-api.yaml
+	# Generate go-bindata, add boilerplate, then cleanup.
+	$(GOBINDATA) -pkg=config -o=$(GOBINDATA_CLUSTERCTL_DIR)/crd_manifests.go $(GOBINDATA_CLUSTERCTL_DIR)/manifest/
+	cat ./hack/boilerplate/boilerplate.generatego.txt $(GOBINDATA_CLUSTERCTL_DIR)/crd_manifests.go > $(GOBINDATA_CLUSTERCTL_DIR)/manifest/crd_manifests.go
+	cp $(GOBINDATA_CLUSTERCTL_DIR)/manifest/crd_manifests.go $(GOBINDATA_CLUSTERCTL_DIR)/crd_manifests.go
+	# Cleanup the manifest folder.
+	$(MAKE) clean-bindata
```

Comment:
If we are basically throwing away the contents of the manifest directory, would it make more sense to use `mktemp` for the destination rather than the current approach of creating and throwing away a directory under the repo?

Reply:
@fabriziopandini mentioned we might have to rework this logic anyways with a different tool, given that bindata is an archived project, I didn't want to put much effort in rewriting this

Reply:
I'm more worried about some cruft accidentally getting committed to the repo if one of the intermediate steps fails.

Reply:
For comparison, if this is at all helpful: https://github.com/kubernetes/kubernetes/blob/master/hack/generate-bindata.sh

Reply:
I'm considering to switch to https://github.com/GeertJohan/go.rice and [append](https://github.com/GeertJohan/go.rice#rice-append-append-resources-to-executable-as-zip-file) the manifest to the binary 
The expected benefit will be to not generate go code (no need of append boilerplate, making it pass/skip lint, ftm etcd). Everything will be part of package/generate  binaries

Reply:
Can we merge this as-is given that it fixes CI for now and we can discuss the rest in a later PR or issue when @fabriziopandini is ready to tackle that?

Reply:
Using .gitignore for this as a temporary fix WFM

Reply:
Done!